cmd/templ/generatecmd/main.go

```diff
@@ -3,17 +3,157 @@ package generatecmd
 import (
 	"context"
 	_ "embed"
+	"flag"
+	"fmt"
+	"io"
 	"log/slog"
+	"regexp"
+	"runtime"
 
 	_ "net/http/pprof"
+
+	"github.com/a-h/templ/cmd/templ/sloghandler"
 )
 
+const generateUsageText = `usage: templ generate [<args>...]
+
+Generates Go code from templ files.
+
+Args:
+  -path <path>
+    Generates code for all files in path. (default .)
+  -f <file>
+    Optionally generates code for a single file, e.g. -f header.templ
+  -stdout
+    Prints to stdout instead of writing generated files to the filesystem.
+    Only applicable when -f is used.
+  -source-map-visualisations
+    Set to true to generate HTML files to visualise the templ code and its corresponding Go code.
+  -include-version
+    Set to false to skip inclusion of the templ version in the generated code. (default true)
+  -include-timestamp
+    Set to true to include the current time in the generated code.
+  -watch
+    Set to true to watch the path for changes and regenerate code.
+  -watch-pattern <regexp>
+    Set the regexp pattern of files that will be watched for changes. (default: '(.+\.go$)|(.+\.templ$)|(.+_templ\.txt$)')
+  -cmd <cmd>
+    Set the command to run after generating code.
+  -proxy
+    Set the URL to proxy after generating code and executing the command.
+  -proxyport
+    The port the proxy will listen on. (default 7331)
+  -proxybind
+    The address the proxy will listen on. (default 127.0.0.1)
+  -notify-proxy
+    If present, the command will issue a reload event to the proxy 127.0.0.1:7331, or use proxyport and proxybind to specify a different address.
+  -w
+    Number of workers to use when generating code. (default runtime.NumCPUs)
+  -lazy
+    Only generate .go files if the source .templ file is newer.
+  -pprof
+    Port to run the pprof server on.
+  -keep-orphaned-files
+    Keeps orphaned generated templ files. (default false)
+  -v
+    Set log verbosity level to "debug". (default "info")
+  -log-level
+    Set log verbosity level. (default "info", options: "debug", "info", "warn", "error")
+  -help
+    Print help and exit.
+
+Examples:
+
+  Generate code for all files in the current directory and subdirectories:
+
+    templ generate
+
+  Generate code for a single file:
+
+    templ generate -f header.templ
+
+  Watch the current directory and subdirectories for changes and regenerate code:
+
+    templ generate -watch
+`
+
+const DefaultWatchPattern = `(.+\.go$)|(.+\.templ$)`
```

Comment:
I think this can be a private const.

Reply:
Oh yeah, I think it can now. Previously it needed to be used elsewhere, but I don't think it does now.